main.go

```diff
@@ -111,8 +116,15 @@ func run(c *cli.Context) error {
 		go func() { startchan <- struct{}{} }()
 		file := c.String("file")
 		configchan := make(chan bool)
+		// XXX Unwatchable Puppet sources should get events from a timer,
+		// with the same interval as the Puppet agent
+		// (hint: puppet config print runinterval)
 		if !c.Bool("no-watch") {
-			configchan = ConfigWatch(file)
+			if c.IsSet("file") {
+				configchan = ConfigWatch(file)
+			} else if strings.HasSuffix(c.String("puppet"), ".pp") {
+				configchan = ConfigWatch(c.String("puppet"))
```

Comment:
Good points however:

1) When we're running with an agent (normal mode) we'll automatically re-run puppet every 30 min. This should be part of the patch.

2) mgmt itself _could_ more generally be run in "puppet mode" where puppet mode in this case means, run every 30 minutes from cron. To do so we use the --converged-timeout flag and we don't most of the of the instant events benefits.